flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/table/FileSystemTableSource.java

```diff
@@ -362,20 +367,29 @@ public TableStats reportStatistics() {
                 TableStats tableStats =
                         ((FileBasedStatisticsReportableInputFormat) bulkReaderFormat)
                                 .reportStatistics(files, producedDataType);
+                if (tableStats.equals(TableStats.UNKNOWN)) {
+                    return tableStats;
+                }
                 return limit == null
                         ? tableStats
                         : new TableStats(Math.min(limit, tableStats.getRowCount()));
             } else if (deserializationFormat instanceof FileBasedStatisticsReportableInputFormat) {
                 TableStats tableStats =
                         ((FileBasedStatisticsReportableInputFormat) deserializationFormat)
                                 .reportStatistics(files, producedDataType);
+                if (tableStats.equals(TableStats.UNKNOWN)) {
+                    return tableStats;
+                }
                 return limit == null
                         ? tableStats
                         : new TableStats(Math.min(limit, tableStats.getRowCount()));
             } else {
                 return TableStats.UNKNOWN;
             }
         } catch (Exception e) {
+            LOG.info(
+                    String.format(
+                            "Reporting statistics failed for file system table source: %s", e));
```

Comment:
LOG.warn(Reporting statistics failed for file system table source: {}", e.getMessage());